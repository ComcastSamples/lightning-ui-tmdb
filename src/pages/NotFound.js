import Page from './Page';
import { TYPOGRAPHY, getHexColor } from '@lightningjs/ui-components';

export default class NotFound extends Page {
  static _template() {
    return {
      _testId: 'NotFound',
      title: 'Not Found',
      ...super._template(),
      SectionsContainer: {
        x: 30,
        y: 30,
        Title: {
          text: {
            ...TYPOGRAPHY.display1,
            text: '404 Page Not Found',
            textColor: getHexColor('#000000'),
          },
        },
      },
    };
  }

  _getFocused() {
    return this;
  }
}
