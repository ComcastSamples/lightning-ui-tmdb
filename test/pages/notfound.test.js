import TestRenderer from '@lightningjs/ui-components/test/lightning-test-renderer';
import NotFound from 'src/pages/NotFound';

describe('NotFound Page', () => {
  let NotFoundPage;
  let testRenderer;
  let page;
  beforeEach(() => {
    NotFoundPage = {
      NotFoundPage: {
        type: NotFound,
      }
    }
    testRenderer = TestRenderer.create(NotFoundPage);
    page = testRenderer.getInstance();
  });

  it('should render', () => {
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
