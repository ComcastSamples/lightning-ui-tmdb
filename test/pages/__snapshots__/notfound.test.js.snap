// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page should render 1`] = `
Object {
  "NotFoundPage": Object {
    "SectionsContainer": Object {
      "active": false,
      "alpha": 1,
      "attached": true,
      "boundsMargin": null,
      "children": Object {
        "Title": Object {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 0,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Title",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": Array [
            "Title",
          ],
          "texture": Object {
            "fontSize": 56,
            "letterSpacing": -0.4,
            "lineHeight": 72,
            "text": "404 Page Not Found",
            "textColor": NaN,
            "type": "TextTexture",
            "verticalAlign": "middle",
          },
          "visible": true,
          "w": 0,
          "x": 0,
          "y": 0,
          "zIndex": 0,
        },
      },
      "clipping": false,
      "color": 4294967295,
      "enabled": true,
      "flex": false,
      "flexItem": false,
      "h": 0,
      "isComponent": undefined,
      "mount": 0,
      "mountX": 0,
      "mountY": 0,
      "pivot": 0.5,
      "pivotX": 0.5,
      "pivotY": 0.5,
      "ref": "SectionsContainer",
      "renderOfScreen": undefined,
      "renderToTexture": false,
      "scale": 1,
      "scaleX": 1,
      "scaleY": 1,
      "state": undefined,
      "tag": [Function],
      "tags": Array [
        "SectionsContainer",
      ],
      "visible": true,
      "w": 0,
      "x": 30,
      "y": 30,
      "zIndex": 0,
    },
  },
}
`;
